coderd/workspaces.go

```diff
@@ -839,6 +885,32 @@ func requestTemplate(ctx context.Context, rw http.ResponseWriter, req codersdk.C
 	return template, true
 }
 
+func claimPrebuild(ctx context.Context, claimer prebuilds.Claimer, db database.Store, logger slog.Logger, req codersdk.CreateWorkspaceRequest, owner workspaceOwner) (*database.Workspace, error) {
+	prebuildsCtx := dbauthz.AsPrebuildsOrchestrator(ctx)
```

Comment:
Is it okay to elevate permissions here? Shouldn't we just rely on user permissions?